### PR TITLE
Month picker

### PR DESCRIPTION
**Pull request by mattsjones**
The long awaited month picker! 

This PR also updates SVGO to add `focusable="true"` to all SVG elements. This solves a problem in IE where SVG's are focusable unless this attribute is present.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -78,10 +80,12 @@
     "less-vars-to-js": "^1.1.1",
     "lodash.debounce": "^4.0.6",
     "lodash.omit": "^4.5.0",
+    "lodash.range": "^3.2.0",
     "mocha": "^2.5.3",
     "mock-css-modules": "^1.0.0",
     "opn": "^4.0.1",
     "opn-cli": "^3.1.0",
+    "pad-left": "^2.1.0",
```
Comment (tamoore) [inline package.json]: LOL 😹

File: react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+          this.blurIfNotFocussed(activeElement);
+        });
+      }
+    }
+  }
+
+  render() {
+    const { value, className, invalid, id } = this.props;
+    const { month, year } = value;
+    const monthValue = month ? `${month}` : '';
```
Comment (mjt01) [inline react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js]: for the same reason [as above](https://github.com/SEEK-Jobs/seek-style-guide/pull/64/files#r87528592), `String(month || '')`? Depending on your thoughts on the previous comment

File: react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    const yearValue = year ? `${year}` : '';
+
+    const rootClasses = classnames({
+      [className]: className,
+      [styles.root]: true
```
Comment (mjt01) [inline react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js]: normally i would flip these that the optional class is applied on top of the default styles. in this case the `root` properties would override the custom properties

File: react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+import range from 'lodash.range';
+
+import Dropdown from '../../Dropdown/Dropdown';
+
+const getYearOptions = () => {
+  const maxYear = new Date().getFullYear();
+  const minYear = maxYear - 100;
+
+  return range(maxYear, minYear - 1).map(value => {
+    const stringValue = `${value}`;
```
Comment (mattsjones) [inline react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js]: Yeah that's probably better as it's more declarative what's it's intending to do

File: react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js (hunk 3 above)
Comment (mattsjones) [inline react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js]: 👍

File: react/fields/MonthPicker/MonthPicker.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+  };
+}
+
+export default class MonthPicker extends Component {
+
+  static displayName = 'MonthPicker';
+
+  static propTypes = {
+    /* eslint-disable consistent-return */
+    id: (props, propName, componentName) => {
```
Comment (tamoore) [inline react/fields/MonthPicker/MonthPicker.js]: I noticed that `Dropdown` uses the same propType validator. I wonder if it's worth extracting these into a shared file that contains common `PropTypes`.

File: react/fields/MonthPicker/MonthPicker.js [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+      if (label && !id) {
+        return new Error(`When ${componentName} has a \`label\`, it should also have an \`id\`.`);
+      }
+    },
+    className: PropTypes.string,
+    /* eslint-enable consistent-return */
+    invalid: PropTypes.bool,
+    label: PropTypes.string,
+    /* eslint-disable consistent-return */
+    labelProps: (props, propName, componentName) => {
```
Comment (tamoore) [inline react/fields/MonthPicker/MonthPicker.js]: This is also in `Dropdown`

File: react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+  static propTypes = {
+    onChange: PropTypes.func,
+    onBlur: PropTypes.func,
+    value: PropTypes.shape({
+      month: PropTypes.number,
+      year: PropTypes.number
+    }),
+    invalid: PropTypes.bool,
+    className: PropTypes.string,
```
Comment (mjt01) [inline react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js]: missing default prop for this

File: react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+
+  static displayName = 'CustomMonthPicker';
+
+  static propTypes = {
+    onChange: PropTypes.func,
+    onBlur: PropTypes.func,
+    value: PropTypes.shape({
+      month: PropTypes.number,
+      year: PropTypes.number
+    }),
```
Comment (mattsjones) [inline react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js]: No

File: react/fields/MonthPicker/MonthPicker.js [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+      ...(id ? { id } : {}),
+      value,
+      onChange,
+      onBlur,
+      invalid
+    };
+
+    return (
+      native ?
+        <NativeMonthPicker {...monthPickerProps} /> :
```
Comment (tamoore) [inline react/fields/MonthPicker/MonthPicker.js]: I got a little confused by the `Native` name, for some reason I thought that meant `ReactNative`. Take from that what you will 😉

File: react/fields/MonthPicker/MonthPicker.test.js [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+  createRenderer
+} from 'react-addons-test-utils';
+import { findAllWithClass } from 'react-shallow-testutils';
+import MonthPicker from './MonthPicker';
+
+chai.use(sinonChai);
+
+const renderer = createRenderer();
+
+describe('MonthPicker', () => {
```
Comment (mjt01) [inline react/fields/MonthPicker/MonthPicker.test.js]: This test suite doesnt test anything about the value of the monthpicker being set. Any thoughts on getting coverage of this?

File: react/fields/MonthPicker/MonthPicker.js [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+function combineClassNames(props = {}, ...classNames) {
+  const { className, ...restProps } = props;
+
+  return {
+    className: classnames.apply(null, [...classNames, className]), // eslint-disable-line no-useless-call
+    ...restProps
+  };
+}
+
+export default class MonthPicker extends Component {
```
Comment (tamoore) [inline react/fields/MonthPicker/MonthPicker.js]: Having looked at this component I wonder whether it needs to be a `composite`. It seems to just be a proxy for the `custom` and the `native` component. What are your thoughts? It could be a functional component? Maybe I'm missing something but it looks like it only renders.

File: react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.js [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+
+  render() {
+    const { value, className, invalid, id, onBlur } = this.props;
+
+    const inputValue = makeMonthString(value);
+
+    const rootClasses = classnames({
+      [styles.root]: true,
+      [className]: className,
+      [styles.invalid]: invalid
```
Comment (mjt01) [inline react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.js]: again, `className` should be last here (and have a `defaultProp`)

File: react/fields/MonthPicker/MonthPicker.test.js (hunk 10 above)
Comment (mattsjones) [inline react/fields/MonthPicker/MonthPicker.test.js]: 🤔  `MonthPicker` simply passed through a `value` & `onChange` function. All we can really test is that it forwards those props. I've done that for onChange & onBlur so I can add one for value as well.

File: react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.less [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+  border-radius: @field-border-radius;
+  color: @sk-charcoal;
+  background-color: @sk-white;
+
+  &::-webkit-inner-spin-button,
+  &::-webkit-calendar-picker-indicator,
+  &::-webkit-clear-button {
+    display: none;
+    -webkit-appearance: none;
+    -moz-appearance: none;
```
Comment (mjt01) [inline react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.less]: is this valid that we are styling webkit pseudo elements with firefox prefix? should we leave this to autoprefixer and make it `appearance: none`?

File: webpack/index.js [hunk 14]
```diff
@@ -92,6 +92,16 @@ const getLocalIdentName = () => isProduction() ?
   '[hash:base64:7]' :
   '__STYLE_GUIDE__[name]__[local]___[hash:base64:7]';
 
+const getSvgoLoader = () => {
+  const svgoConfig = JSON.stringify({
+    plugins: [
+      { addAttributesToSVGElement: { attribute: 'focusable="false"' } }
```
Comment (mjt01) [inline webpack/index.js]: 👏

File: react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.less (hunk 13 above)
Comment (mattsjones) [inline react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.less]: I'm checking on autoprefixer, I was under the impression appearance was more specific and wouldn't be auto-prefixed. However, I do need to move the `-moz-apearance` out of this selector.

File: react/fields/MonthPicker/MonthPicker.js (hunk 11 above)
Comment (mattsjones) [inline react/fields/MonthPicker/MonthPicker.js]: 🤔  Interesting. The only thing is there is a nice consistency between all the fields in style guide atm in the way they're set out. Personally I found this way more readable for a component as large as this (although not that complex). Value in both approaches to me. @mjt01 thoughts?

File: react/fields/MonthPicker/MonthPicker.js (hunk 5 above)
Comment (mattsjones) [inline react/fields/MonthPicker/MonthPicker.js]: You're right. There are a lot of duplicates in style guide atm. Also we want to pull label and message (and any other common patterns) into reusable components so we're not duplicating so much. Feel both of these would be better off having a dedicated PR though.

File: react/fields/MonthPicker/MonthPicker.js (hunk 9 above)
Comment (tamoore) [inline react/fields/MonthPicker/MonthPicker.js]: I'm terrible at naming! It's like my endless achilles heel and "just when I think I'm out, it pulls me back in".  Soooo....maybe it's better to name it for what the target is current, i.e. mobile devices. So something like, `StandardMobileMonthPicker`. Just a thought.

Comment (tamoore) [inline react/fields/MonthPicker/MonthPicker.js]: Or `TouchDeviceMonthPicker`

Comment (mjt01) [inline react/fields/MonthPicker/MonthPicker.js]: Yeah as im trying to think of a better name i lean more to `Native` as it really is cross platform/device etc.

Comment (mattsjones) [inline react/fields/MonthPicker/MonthPicker.js]: I WIN 🏆

File: react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.js [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+          className={styles.input}
+          type="month"
+          value={inputValue}
+          onChange={this.handleChange}
+          onBlur={onBlur}
+        />
+      </div>
+    );
+  }
+}
```
Comment (tamoore) [inline react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.js]: 👍

File: react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.less [hunk 16]
```diff
@@ -0,0 +1,59 @@
+@import (reference) "~seek-style-guide/theme";
```
Comment (tamoore) [inline react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.less]: I'm not sure if this is mean to be there but I get a strange chevron on iPad

Comment (tamoore) [inline react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.less]: screen shot 2016-11-11 at 3 32 43 pm

File: react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js [hunk 17]
```diff
@@ -0,0 +1,169 @@
...
+    onChange({
+      month,
+      year: parseInt(value, 10)
+    });
+  }
+
+  blurIfNotFocussed(active) {
+    const { onBlur, value } = this.props;
+
+    if (active !== this.monthInput && active !== this.yearInput) {
```
Comment (mjt01) [inline react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js]: `&& typeof onBlur === 'function'`

File: react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js [hunk 18]
```diff
@@ -0,0 +1,169 @@
...
+      onBlur(value);
+    }
+  }
+
+  handleBlur({ relatedTarget }) {
+    const { onBlur } = this.props;
+
+    if (onBlur) {
```
Comment (mjt01) [inline react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js]: ah gotcha, maybe check its a function too

Comment (mattsjones) [inline react/fields/MonthPicker/CustomMonthPicker/CustomMonthPicker.js]: 👍

File: react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.less [hunk 19]
```diff
@@ -16,13 +16,13 @@
   border-radius: @field-border-radius;
   color: @sk-charcoal;
   background-color: @sk-white;
+  appearance: none;
 
   &::-webkit-inner-spin-button,
   &::-webkit-calendar-picker-indicator,
   &::-webkit-clear-button {
     display: none;
     -webkit-appearance: none;
```
Comment (mjt01) [inline react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.less]: this is redundant with `appearance: none` being outside this pseudo selector. confirming that it still works as expected if its on the class not the pseudo element?

Comment (mattsjones) [inline react/fields/MonthPicker/NativeMonthPicker/NativeMonthPicker.less]: It actually needed to be moved outside to deal with a separate issue on iOS. This combination of styles should be good now. The `appearance:none` takes care of the normal scenarios and the pseudo selectors deal with the extra webkit specific parts. Both are required though.

**Comment by mjt01** [comment]
:art:
